src/utilities/xml/Validator.cpp

```diff
@@ -0,0 +1,143 @@
+*  derived from this software without specific prior written permission from the respective party.
+*
+*  (4) Other than as required in clauses (1) and (2), distributions in any form of modifications or other derivative works
+*  may not use the "OpenStudio" trademark, "OS", "os", or any other confusingly similar designation without specific prior
+*  written permission from Alliance for Sustainable Energy, LLC.
+*
+*  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDER(S) AND ANY CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+*  INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+*  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER(S), ANY CONTRIBUTORS, THE UNITED STATES GOVERNMENT, OR THE UNITED
+*  STATES DEPARTMENT OF ENERGY, NOR ANY OF THEIR EMPLOYEES, BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+*  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF
+*  USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+*  STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+*  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+***********************************************************************************************************************/
+
+#include "Validator.hpp"
+
+#include <xercesc/util/XMLString.hpp>
+#include <xercesc/parsers/XercesDOMParser.hpp>
+#include <xercesc/framework/LocalFileInputSource.hpp>
+#include <xercesc/sax/ErrorHandler.hpp>
+#include <xercesc/sax/SAXParseException.hpp>
+#include <xercesc/validators/common/Grammar.hpp>
+
+namespace openstudio {
+
+class ParserErrorHandler : public xercesc::ErrorHandler
+{
+ private:
+  void reportParseException(const xercesc::SAXParseException& ex) {
+    char* msg = xercesc::XMLString::transcode(ex.getMessage());
+    fprintf(stderr, "at line %llu column %llu, %s\n", ex.getLineNumber(), ex.getColumnNumber(), msg);
+    xercesc::XMLString::release(&msg);
+  }
+
+ public:
+  void warning(const xercesc::SAXParseException& ex) {
+    reportParseException(ex);
+  }
+
+  void error(const xercesc::SAXParseException& ex) {
+    reportParseException(ex);
+  }
+
+  void fatalError(const xercesc::SAXParseException& ex) {
+    reportParseException(ex);
+  }
+
+  void resetErrors() {}
+};
+
+Validator::Validator(const openstudio::path& xsdPath) : m_xsdPath(openstudio::filesystem::system_complete(xsdPath)) {
+  if (!openstudio::filesystem::exists(xsdPath)) {
+    LOG_AND_THROW("'" << toString(xsdPath) << "' does not exist");
+  } else if (!openstudio::filesystem::is_regular_file(xsdPath)) {
+    LOG_AND_THROW("'" << toString(xsdPath) << "' XSD cannot be opened");
+  }
+
+  xercesc::XMLPlatformUtils::Initialize();
+
+  // create the parser here?
```

Comment:
If we stick to a Ctor that takes the schema (and given there isn't a setXsdPath), then I would say yes, create the parser in the ctor.

Reply:
Additionally, make xercesc::XercesDOMParser parser; a member variable.